packages/vsx-registry/src/node/vsx-cli.ts

```diff
@@ -0,0 +1,38 @@
+// *****************************************************************************
+// Copyright (C) 2023 Ericsson and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { CliContribution } from '@theia/core/lib/node';
+import { injectable } from '@theia/core/shared/inversify';
+import { Argv } from '@theia/core/shared/yargs';
+import { OVSXRouterConfig } from '@theia/ovsx-client';
+import * as fs from 'fs';
+
+@injectable()
+export class VsxCli implements CliContribution {
+
+    ovsxRouterConfig: OVSXRouterConfig | undefined;
+
+    configure(conf: Argv<{}>): void {
+        conf.option('ovsx-router-config', { description: 'JSON configuration file for the OVSX router client', type: 'string' });
```

Comment:
Yeah I wish we abstracted away the way to configure such things so that 1) we don't directly depend on the yargs APIs and 2) when defining a CLI option it would also look at the environment automatically.

I can see to align both options.